### ISSUE TITLE
Support all container events in TestExecutionSummary

### ISSUE THREAD

**Issue by mfulton26** [junit-team/junit5#420]
With the below code I get the following output:

```
Test run finished after 35 ms
[         7 tests found      ]
[         0 tests skipped    ]
[         3 tests started    ]
[         0 tests aborted    ]
[         3 tests successful ]
[         0 tests failed     ]
[         1 containers failed]
```

But now I don't know how many containers were found/skipped/started/aborted/successful and my number of tests found is higher than expected. I'd like to get output like this:

```
Test run finished after 31 ms
[         3 tests found          ]
[         0 tests skipped        ]
[         3 tests started        ]
[         0 tests aborted        ]
[         3 tests successful     ]
[         0 tests failed         ]
[         4 containers found     ]
[         1 containers skipped   ]
[         3 containers started   ]
[         1 containers aborted   ]
[         1 containers successful]
[         1 containers failed    ]
```
---
```java
package demo;

import org.junit.jupiter.api.Test;
import org.junit.platform.engine.*;
import org.junit.platform.engine.support.descriptor.AbstractTestDescriptor;
import org.junit.platform.engine.support.descriptor.EngineDescriptor;

public class DemoTestEngine implements TestEngine {

    private UniqueId rootUniqueId;

    @Override
    public String getId() {
        return "demo";
    }

    @Override
    public TestDescriptor discover(EngineDiscoveryRequest discoveryRequest, UniqueId uniqueId) {
        EngineDescriptor engineDescriptor = new EngineDescriptor(uniqueId, "demo");
        RootTestDescriptor root = new RootTestDescriptor(uniqueId.append("root", "0"), "root #0");
        engineDescriptor.addChild(root);
        rootUniqueId = root.getUniqueId();
        return engineDescriptor;
    }

    @Override
    public void execute(ExecutionRequest request) {
        EngineExecutionListener engineExecutionListener = request.getEngineExecutionListener();
        TestDescriptor engineDescriptor = request.getRootTestDescriptor();

        engineExecutionListener.executionStarted(engineDescriptor);

        TestDescriptor root = engineDescriptor.findByUniqueId(rootUniqueId).get();

        engineExecutionListener.executionStarted(root);

        TestDescriptor container = new ContainerTestDescriptor(rootUniqueId.append("container", "1"), "container #1");
        root.addChild(container);

        engineExecutionListener.dynamicTestRegistered(container);
        engineExecutionListener.executionStarted(container);

        UniqueId containerUid = container.getUniqueId();

        TestDescriptor dynamicTest1 = new TestTestDescriptor(containerUid.append("test", "1"), "dynamic test #1");
        container.addChild(dynamicTest1);
        engineExecutionListener.dynamicTestRegistered(dynamicTest1);
        engineExecutionListener.executionStarted(dynamicTest1);
        engineExecutionListener.executionFinished(dynamicTest1, TestExecutionResult.successful());

        TestDescriptor dynamicTest2 = new TestTestDescriptor(containerUid.append("test", "2"), "dynamic test #2");
        container.addChild(dynamicTest2);
        engineExecutionListener.dynamicTestRegistered(dynamicTest2);
        engineExecutionListener.executionStarted(dynamicTest2);
        engineExecutionListener.executionFinished(dynamicTest2, TestExecutionResult.successful());

        TestDescriptor subcontainer = new ContainerTestDescriptor(containerUid.append("container", "2"), "container #2");
        container.addChild(subcontainer);
        engineExecutionListener.dynamicTestRegistered(subcontainer);
        engineExecutionListener.executionStarted(subcontainer);

        UniqueId subcontainerUid = subcontainer.getUniqueId();

        TestDescriptor dynamicTest3 = new TestTestDescriptor(subcontainerUid.append("test", "3"), "dynamic test #3");
        subcontainer.addChild(dynamicTest3);
        engineExecutionListener.dynamicTestRegistered(dynamicTest3);
        engineExecutionListener.executionStarted(dynamicTest3);
        engineExecutionListener.executionFinished(dynamicTest3, TestExecutionResult.successful());

        TestDescriptor skippedSubcontainer = new ContainerTestDescriptor(containerUid.append("container", "3"), "container #3");
        container.addChild(skippedSubcontainer);
        engineExecutionListener.dynamicTestRegistered(skippedSubcontainer);
        engineExecutionListener.executionSkipped(skippedSubcontainer, "reason");

        TestDescriptor abortedSubcontainer = new ContainerTestDescriptor(containerUid.append("container", "4"), "container #4");
        container.addChild(abortedSubcontainer);
        engineExecutionListener.dynamicTestRegistered(abortedSubcontainer);
        engineExecutionListener.executionStarted(abortedSubcontainer);
        engineExecutionListener.executionFinished(abortedSubcontainer, TestExecutionResult.aborted(new IllegalStateException("message")));

        engineExecutionListener.executionFinished(subcontainer, TestExecutionResult.failed(new IllegalStateException("sad")));

        engineExecutionListener.executionFinished(container, TestExecutionResult.successful());

        engineExecutionListener.executionFinished(root, TestExecutionResult.successful());

        engineExecutionListener.executionFinished(engineDescriptor, TestExecutionResult.successful());
    }

    static class RootTestDescriptor extends AbstractTestDescriptor {
        RootTestDescriptor(UniqueId uniqueId, String displayName) {
            super(uniqueId, displayName);
        }

        @Override
        public boolean isContainer() {
            return true;
        }

        @Override
        public boolean isTest() {
            return false;
        }

        @Override
        public boolean hasTests() {
            return true;
        }
    }

    static class ContainerTestDescriptor extends AbstractTestDescriptor {
        ContainerTestDescriptor(UniqueId uniqueId, String displayName) {
            super(uniqueId, displayName);
        }

        @Override
        public boolean isContainer() {
            return true;
        }

        @Override
        public boolean isTest() {
            return false;
        }
    }

    static class TestTestDescriptor extends AbstractTestDescriptor {
        TestTestDescriptor(UniqueId uniqueId, String displayName) {
            super(uniqueId, displayName);
        }

        @Override
        public boolean isContainer() {
            return false;
        }

        @Override
        public boolean isTest() {
            return true;
        }
    }
}
```

**Comment by sbrannen** [comment]
Thanks for raising the issue.

Initial, limited, support for reporting on containers was added in #402.

So we'll pick up from there in a later milestone.

**Comment by sbrannen** [comment]
FYI: I have updated the title of this issue accordingly.

**Comment by sbrannen** [comment]
_in progress_

**Comment by sbrannen** [comment]
This has been addressed in 7e14f9484c26ce347d376c3f28d22b344daba2b5.

So feel free to try out the next snapshot and provide feedback.

cheers!
